src/components/Home.js

```diff
@@ -0,0 +1,161 @@
+    }
+    componentDidMount(){
+      this.props.getOffers();
+      this.props.getCategories();
+    }
+    static getDerivedStateFromProps(props,state){
+      if(props.homeApi.bannerData && props.homeApi.bannerdata_searching_success){
+        state.bannerData = props.homeApi.bannerData;
+        state.currentInstance.forceCheckSliderImage();
+      }
+      if(props.homeApi.categoryData && props.homeApi.categorydata_searching_success){
+        state.categoryData = props.homeApi.categoryData;
+      }
+      return state;
+    }
+    forceCheckSliderImage=()=>{
+      forceCheck();
+    }
+    doResizeActions=()=>{
+      let dimensions = getViewPortDimensions();
+      this.setState({dimensions: dimensions});
+    }
+  checkoutComplete = () => {
+    this.props.resetCartData();
+    this.onCloseModal();
+    toastr.success(Constants.TEXTS.TOASTS.cart_emptied,Constants.TEXTS.TOASTS.order_placed,{timeOut:1000});
+  }
+  showCartView = () => {
+    this.setState({ openModal: true });
+    document.getElementsByTagName('body')[0].style.overflow = 'hidden';
+    document.getElementsByTagName('html')[0].style.overflow = 'hidden';
+  }
+
+  onCloseModal = () => {
+    this.setState({ openModal: false });
+    document.getElementsByTagName('body')[0].style.overflow = 'auto';
+    document.getElementsByTagName('html')[0].style.overflow = 'auto';
+  };
+    navigateToPlp=(e,data)=>{
+      if(data)
+      this.props.router.push('/products?cat='+data.name+'&id='+data.id);
+    }
+  render(){
+    const settings = {
+      dots: true
+    };
+    const {homeApi,productInfo,router} = this.props;
+    const {bannerData,categoryData,openModal} = this.state;
+  return (
+    <main className="App home-page">
+      <div className="overlay" style={{display: (homeApi.bannerdata_searching || homeApi.categorydata_searching) ? "block" : "none"}}>
+            <div className="loading"><SvgLoading /></div>
+      </div>
+      <Header router={router} cartClick={this.showCartView} cartInfo={productInfo.cartItems}></Header>
+      <section className="offerArea content-seperator">
+        <div className="slider-container">
+          <Slider {...settings}>
+            {bannerData && bannerData.length ? bannerData.map((data, idx) => (
+              <div key={idx}>
+                <LazyLoad height={200} once>
+                  <img data-id={data.id} src={window.location.origin + data.bannerImageUrl} alt={data.bannerImageAlt} />
+                </LazyLoad>
+              </div>
+            )) : null}
+          </Slider>
+        </div>
+      </section>
+      <div className="category-banners-area">
+          {categoryData && categoryData.length ? categoryData.filter((item)=>{return item.order>= 0}).map((data,i)=>(<section className="single-banner-area content-seperator" key={i}>
+            {data.imageUrl && (<div>
+            <div className="col span-1-of-2">
+              {i%2 != 0 ? (<div className="row login-left-area">
```

Comment:
try extracting it into a separate component and use that component instead

Reply:
done